pytorch_lightning/utilities/fetching.py

```diff
@@ -320,9 +330,11 @@ def move_to_device(self, batch: Any) -> Any:
 
     def on_fetch_start(self) -> "torch.cuda.Event":
         # create a cuda event used to record the async stream of data to device.
+        self._start_profiler()
         return torch.cuda.Event()
```

Comment:
Probably overkill but if creating the `cuda.Event()` was a very expensive call, then we would want to do

```suggestion
        event = torch.cuda.Event()
        self._start_profiler()
        return event
```